Automattic/jetpack
# CRM: Fix tax rate deletion for single entries on tax settings page.

## The issue:

* Currently (since CRM v5.1.0), it isn't possible to delete the last remaining tax rate on the Tax Settings page - `/wp-admin/admin.php?page=zerobscrm-plugin-settings&tab=tax`.
* The issue was that if there was only one entry and we try to delete it, `$_POST['jpcrm-taxtable-line']` has no value (see [`admin/settings/tax.page.php`](https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/crm/admin/settings/tax.page.php))
* As such, with the refactoring of the code that [happened here](https://github.com/Automattic/zero-bs-crm/pull/2201/files) and was live from CRM v 5.1.0, we now return quite early on ([at this stage](https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/crm/admin/settings/tax.page.php#L23)). Returning this early meant the deletion that [took place here,](https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/crm/admin/settings/tax.page.php#L67) wasn't happening.


## Proposed changes:

* This PR changes the order in which POST checks occur (specifically for` $_POST['jpcrm-taxtable-line']`), to allow for the deletion of tax rates when there is only one tax rate in the tax rate list, on the Tax Settings page.



### Other information:


## Jetpack product discussion

https://github.com/Automattic/zero-bs-crm/issues/2753

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

* To replicate the issue, on a site running CRM version 5.1.0 or higher, visit the CRM Settings > Tax Settings page - `/wp-admin/admin.php?page=zerobscrm-plugin-settings&tab=tax` - and make sure you have just one tax rate entry. 
* Try to delete that entry and save the settings - you should see a blank Tax Settings page after this. On page refresh, the data remains.
* To test this fix, using the Jetpack Beta plugin on a test site with this branch applied or testing locally with the branch checked out, visit the CRM Settings > Tax Settings page: `/wp-admin/admin.php?page=zerobscrm-plugin-settings&tab=tax`.
* Make sure you just have one entry, and try to delete that (make sure to click the Save Settings button).
* The page should refresh and the tax rate should be gone.
* Experiment with adding and deleting multiple tax rates - there should be no issues.

*Current result after deleting the last entry and clicking 'Save Settings' (and the new rate remains after refreshing the page:*

<img width="1250" alt="Screenshot 2023-03-01 at 2 25 53 pm" src="https://user-images.githubusercontent.com/16754605/222168731-79d56e66-7cb0-4a7f-8e65-47df6378f666.png">

*Expected result with PR applied after deleting the last entry and clicking 'Save Settings':*

<img width="1253" alt="Screenshot 2023-03-01 at 2 25 26 pm" src="https://user-images.githubusercontent.com/16754605/222168759-93a7e9d0-468b-4199-b355-b50bcb28ddb5.png">
